design/ceph/object/store.md

```diff
@@ -33,6 +34,10 @@ spec:
     port: 80
     securePort: 443
     instances: 3
+    hosting:
+      dnsNames:
+      - "http://rook-ceph-rgw-my-store.rook-ceph.svc:80"
+      - "http://my-ingress.mydomain.com"
```

Comment:
We need some clarification here. I haven't seen any ceph docs that suggest that the DNS name should have `http(s)://` or a port spec. I have only seen `x.y.z`, and that's it. Where are you getting information that DNS names can have `http://` or a port? 

Here: https://access.redhat.com/documentation/en-us/red_hat_ceph_storage/2/html/object_gateway_guide_for_ubuntu/configuration#setting_up_the_gateway

Here: https://docs.ceph.com/en/latest/radosgw/s3/commons/#bucket-and-host-name

Here: https://documentation.suse.com/ses/7/html/ses-all/cha-ceph-gw.html#ogw-s3-interface-access

Here: https://www.ibm.com/docs/en/storage-ceph/6?topic=configuration-add-wildcard-dns

Reply:
right. this is just a dns name, not a url with protocol or port number

Reply:
@thotz please update the design doc to note this. 

We might want to also do input validation to ensure that this only has valid hostname characters. This is codified by RFC-1123, and there is a k8s apimachinery library to validate values: `IsDNS1123Subdomain()`

Reply:
I don't see anything in the design about validating inputs to valid DNS-1123 hostnames, or mention that it needs to be communicated to users in the docs when implemented.